pkgs/development/compilers/gcc/ng/common/gcc/default.nix

```diff
@@ -0,0 +1,249 @@
+{
+  lib,
+  stdenv,
+  gcc_meta,
+  release_version,
+  version,
+  monorepoSrc ? null,
+  langAda ? false,
+  langC ? true,
+  langCC ? true,
+  langD ? false,
+  langFortran ? false,
+  langGo ? false,
+  langJava ? false,
+  langObjC ? stdenv.targetPlatform.isDarwin,
+  langObjCpp ? stdenv.targetPlatform.isDarwin,
+  langJit ? false,
+  enablePlugin ? lib.systems.equals stdenv.hostPlatform stdenv.buildPlatform,
+  runCommand,
+  buildPackages,
+  isl,
+  zlib,
+  gmp,
+  libmpc,
+  mpfr,
+  libelf,
+  perl,
+  texinfo,
+  which,
+  gettext,
+  getVersionFile,
+  buildGccPackages,
+  targetPackages,
+  libc,
+  bintools,
+}:
+let
+  inherit (stdenv) targetPlatform hostPlatform;
+  targetPrefix = lib.optionalString (targetPlatform != hostPlatform) "${targetPlatform.config}-";
```

Comment:
@RossComputerGuy A bit late to the party, but in https://github.com/NixOS/nixpkgs/pull/389469 I fixed a somewhat obscure cross-compilation bug by using `lib.systems.equals` consistently everywhere for gcc, but in this ng version some direct equality checks are present again. I think it would be good to keep things consistent from the start. I can make PR for this, but maybe it's easier if you do it while you're still fixing things?